packages/slider/index.tsx

```diff
@@ -0,0 +1,677 @@
+    if (disabled) {
+      return;
+    }
+
+    this.preventFocusState = true;
+    this.setInTransit(!this.handlingThumbTargetEvt);
+    this.handlingThumbTargetEvt = false;
+    this.setActive(true);
+
+    const moveHandler = (moveEvent: MouseLikeEvent) => {
+      this.handleMove(moveEvent);
+    };
+
+    const moveEventType = MOVE_EVENT_MAP[downEvent.type as DownEventType];
+
+    // Note: upHandler is [de]registered on ALL potential pointer-related release event types, since some browsers
+    // do not always fire these consistently in pairs.
+    // (See https://github.com/material-components/material-components-web/issues/1192)
+    const upHandler = () => {
+      this.handleUp();
+      this.adapter.deregisterBodyInteractionHandler(moveEventType, moveHandler);
+      UP_EVENTS.forEach((evtName) =>
+        this.adapter.deregisterBodyInteractionHandler(evtName, upHandler)
+      );
+    };
+
+    this.adapter.registerBodyInteractionHandler(moveEventType, moveHandler);
+    UP_EVENTS.forEach((evtName) =>
+      this.adapter.registerBodyInteractionHandler(evtName, upHandler)
+    );
+    this.setValueFromEvt(downEvent);
+  };
+
+  /**
+   * Called when the user's interaction with the slider ends
+   */
+  private handleUp() {
+    this.setActive(false);
+    this.adapter.notifyChange();
+  }
+
+  /**
+   * Called when the user moves the slider
+   */
+  private handleMove(evt: MouseLikeEvent) {
+    evt.preventDefault();
+    this.setValueFromEvt(evt);
+  }
+
+  /**
+   * Returns the pageX of the event
+   */
+  private getPageX(evt: MouseLikeEvent | ReactMouseLikeEvent): number {
+    if (
+      (evt as TouchEvent).targetTouches &&
+      (evt as TouchEvent).targetTouches.length > 0
+    ) {
+      return (evt as TouchEvent).targetTouches[0].pageX;
+    }
+    return (evt as MouseEvent).pageX;
+  }
+
+  /**
+   * Sets the slider value from an event
+   */
+  private setValueFromEvt(evt: MouseLikeEvent | ReactMouseLikeEvent) {
+    const pageX = this.getPageX(evt);
+    const value = this.computeValueFromPageX(pageX);
+    this.setValue(value, true);
+  }
+
+  /**
+   * Computes the new value from the pageX position
+   */
+  private computeValueFromPageX(pageX: number): number {
+    const {max, min} = this.props;
+    const xPos = pageX - this.rect.left;
+    let pctComplete = xPos / this.rect.width;
+    if (this.adapter.isRTL()) {
+      pctComplete = 1 - pctComplete;
+    }
+    // Fit the percentage complete between the range [min,max]
+    // by remapping from [0, 1] to [min, min+(max-min)].
+    return min + pctComplete * (max - min);
+  }
+
+  /**
+   * Handles keydown events
+   */
+  onKeyDown = (evt: React.KeyboardEvent) => {
+    const keyId = this.getKeyId(evt);
+    const value = this.getValueForKeyId(keyId);
+    if (isNaN(value)) {
+      return;
+    }
+
+    // Prevent page from scrolling due to key presses that would normally scroll the page
+    evt.preventDefault();
+    this.adapter.addClass(cssClasses.FOCUS);
```

Comment:
You should most likely be calling foundation.handleDown_. I know this is a private method, however you can refer to  https://github.com/material-components/material-components-web/issues/2813 in a comment when using a private event handler method from the foundation in this slider package.